### PR TITLE
Update packaging

### PR DESCRIPTION
**Pull request by benjeffery**
Bring packaging in line with other tskit repos - I've also made the changes for PyPI uploading but this won't work until I get permissions at PyPI.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -44,6 +44,29 @@ jobs:
...
+            rm dist/*
+            python -m build
+
+      - run:
+          name: Install from the distribution tarball
+          command: |
+            python -m venv venv
+            source venv/bin/activate
+            pip install dist/*.tar.gz
+            python -c 'import tsinfer; print(tsdate.__version__)'
```
Comment (jeromekelleher) [inline .circleci/config.yml:64]: tsinfer, here and below

File: .github/workflows/docker/buildwheel.sh [hunk 2]
```diff
@@ -13,9 +13,12 @@ git config --global --add safe.directory /project
 # Fetch the full history as we'll be missing tags otherwise.
 git fetch --unshallow
```
Comment (jeromekelleher) [inline .github/workflows/docker/buildwheel.sh:14]: tsdate is Python only right, so can we ditch the docker stuff entirely?

**Review by jeromekelleher — COMMENTED** [review]
LGTM - let's ditch the docker manylinux build if we can though.

File: .circleci/config.yml (hunk 1 above)
Comment (benjeffery) [reply .circleci/config.yml]: Fixed

File: .github/workflows/docker/buildwheel.sh (hunk 2 above)
Comment (benjeffery) [reply .github/workflows/docker/buildwheel.sh]: Yep, removed.

**Comment by benjeffery** [comment]
Think this is good to go!

**Comment by benjeffery** [comment]
@hyanwong Can I get a review here? Would like to merge if poss.

**Review by hyanwong — APPROVED** [review]
LGTM. Not sure where the `author_email` and `description` etc fields go now, but assume it's all fixed/

**Comment by benjeffery** [comment]
Yikes, thanks for that, I hadn't added `setup.cfg` to the PR! Should be there now.

**Pull request merged**